### DISCUSSION TITLE
Reset intercepted requests

### DISCUSSION THREAD

**Discussion by d-koppenhagen** [cypress-io/cypress#17051] [Questions and Help] [answered]
Hey y'all,

Is there a way to reset an intercepted request?
Background:
I have a lot's of test files using almost the same stuff inside `beforeEach`. So I decided to outsource all that stuff in a separate reusable function:

```js
export const setupWithMocks = () => {
  // mock common routes
  cy.intercept('/foo', { fixture: 'foo.json' });
  cy.intercept('/bar', { fixture: 'bar.json' });
  cy.intercept('/baz', { fixture: 'baz.json' });

  // reset / set localStorage
  window.localStorage.removeItem(`userConfig`);
  window.localStorage.setItem('foo', ['bar']);

  // ... and some more
};
```

This work perfectly for lots of tests:

```js
describe(`...`, () => {
  beforeEach(() => {
    setupWithMocks();
  });

  // ...
});
```

But now I want to NOT intercept one of the requests just for a single test case:

```js
describe(`...`, () => {
  beforeEach(() => {
    setupWithMocks();
  });

  it('should display the map view', () => {
    // ??? how can I reset / remove the interception for e.g. cy.intercept('/foo', { fixture: 'foo.json' });
  });

  // ...
});
```

How can I do this?
When I checke dout the cypress documentation for `intecept` I couldn't find any hint how to reset / unstub the intercepted call.
https://docs.cypress.io/api/commands/intercept

Any ideas? Is this maybe a missing feature?

**Comment by flotwig** [comment, marked as answer]
The way to do this is to define a more specific `cy.intercept()` override that continues the response without stubbing it. Since non-`middleware` `cy.intercept()`s are matched [from newest to oldest](https://docs.cypress.io/api/commands/intercept#Interception-lifecycle), this will work:

```ts
cy.intercept('/foo', (req) => {
	// override the previously-declared stub to just continue the request instead of stubbing
	req.continue()
})
```

**Reply by wereHamster** [reply]
That reads more like a workaround. How about something like this:

```js
const interceptor = cy.intercept(…)

cy.stuff()

interceptor.dispose()
```

**Reply by TrevinAvery** [reply]
> The way to do this is to define a more specific `cy.intercept()` override that continues the response without stubbing it. Since non-`middleware` `cy.intercept()`s are matched [from newest to oldest](https://docs.cypress.io/api/commands/intercept#Interception-lifecycle), this will work:
> 
> ```ts
> cy.intercept('/foo', (req) => {
> 	// override the previously-declared stub to just continue the request instead of stubbing
> 	req.continue()
> })
> ```

This is not a solution. Calling `continue()` will skip all other intercepts. It is quite possible that there are several intercepts, and you only want to remove one of them. Having a mechanism to remove or disable a specific intercept would be very useful.

**Reply by SalahAdDin** [reply]
> > The way to do this is to define a more specific `cy.intercept()` override that continues the response without stubbing it. Since non-`middleware` `cy.intercept()`s are matched [from newest to oldest](https://docs.cypress.io/api/commands/intercept#Interception-lifecycle), this will work:
> > ```ts
> > cy.intercept('/foo', (req) => {
> > 	// override the previously-declared stub to just continue the request instead of stubbing
> > 	req.continue()
> > })
> > ```
> 
> This is not a solution. Calling `continue()` will skip all other intercepts. It is quite possible that there are several intercepts, and you only want to remove one of them. Having a mechanism to remove or disable a specific intercept would be very useful.

It seems there is no mechanism yet, right?
